app.js

```diff
@@ -18,11 +23,260 @@ const configureServer = (httpServer) => {
+const client = new MongoClient(uri);
+// Connexion à MongoDB
+client.connect()
+  .then(() => {
+    console.log('Connecté à MongoDB');
+    // Définition de la collection "users"
+    db = client.db('evaluetonsavoir');
+    usersCollection = db.collection('users');
+    foldersCollection = db.collection('folders');
+  })
+  .catch(err => console.error('erreur de connexion MongoDB :', err));
+
+const transporter = nodemailer.createTransport({
+  service: 'gmail',
+  auth: {
+    user: 'infoevaluetonsavoir@gmail.com', // Adresse courriel Gmail utilisée pour l'envoi
+    pass: 'vvml wmfr dkzb vjzb' // Mot de passe de votre adresse courriel Gmail
+  }
+});
 
 setupWebsocket(io);
 app.use(cors());
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
 
 const port = process.env.PORT || 4400;
 server.listen(port, () => {
   console.log(`Server is running on http://localhost:${port}`);
+}); // connection au port  du serveur
+
+//Fonction pour enregistrer un nouvel usager 
+app.post('/register', async (req, res) => {
+
+  if (!req.body || !req.body.email || !req.body.password) {
+    return res.status(400).json({ message: 'Il manque un courriel ou un mot de passe' });
+  }
+
+  const { email, password } = req.body;
+  const existingUser = await usersCollection.findOne({ email });
+
+  if (existingUser) {
+    return res.status(400).json({ message: 'L\'utilisateur existe déjà' });
+  }
+  //hashage du mot de passe
+  const hashedPassword = await bcrypt.hash(password, 10);
+  //ajout de l'usager
+  await usersCollection.insertOne({ email, password: hashedPassword });
+
+  //envoi d'un courriel à l'usager pour confirmer la création du compte  
+  transporter.sendMail({
+    from: 'infoevaluetonsavoir@gmail.com',
+    to: email,
+    subject: 'Confirmation de compte',
+    text: 'Votre compte a été créé avec succès.'
+  });
+
+  res.status(201).json({ message: 'Utilisateur créé avec succès' });
+});
+
+//Fonction pour la connexion
+app.post('/login', async (req, res) => {
+
+  if (!req.body || !req.body.email || !req.body.password) {
+    return res.status(400).json({ message: 'Il manque un courriel ou un mot de passe' });
+  }
+
+  const { email, password } = req.body;
+  //Trouver l'usager dans la BD
+  const user = await usersCollection.findOne({ email });
+
+  if (!user) {
+    return res.status(404).json({ message: 'l\'utilisateur ou le mot de passe son incorrect' });
+  }
+  //Vérification de la correspondance du mot de passe 
+  const passwordMatch = await bcrypt.compare(password, user.password);
+  if (!passwordMatch) {
+    return res.status(401).json({ message: 'l\'utilisateur ou le mot de passe son incorrect' });
+  }
+
+  res.status(200).json({ message: 'Connexion réussie' });
+});
+
+//Fonction pour réinitialiser le mot de passe
+app.post('/reset-password', async (req, res) => {
+  const { email } = req.body;
+  const newPassword = Math.random().toString(36).slice(-8); // Générer un nouveau mot de passe temporaire
```

Comment:
Il faudrait voir si c'est encore à prescrire comme façon de réinitialiser les mots de passes. Je crois qu'un lien avec une page de reset est le standard mainenant. Ça va aussi nous éviter de gérer le mot de passe temporaire qu'on doit faire expirer après 24h...

J'ai trouvé ça et il y a plein d'info qui dit qu'un lien est meilleur.
https://ux.stackexchange.com/questions/72358/should-password-reset-functionality-email-a-temporary-password-or-just-a-reset-l